### DISCUSSION TITLE
Codespaces instability making it impossible to use

### DISCUSSION THREAD

**Discussion by talk2MeGooseman** [community/community#3847] [Codespaces] [answered]
For the past couple of weeks, Codespaces has been impossible to work in. My already existing Codespace started to lose connection and attempt to reconnect constantly. To double-check that I didn't somehow mess up my container I went ahead and deleted my codespace and re-created it with the same issue. 

Here is a screenshot of one of the logs:
![image](https://user-images.githubusercontent.com/1203718/115969061-e7eadd00-a4ef-11eb-8676-081053361af8.png)

I have also seen a couple of other issues in this discussion board with people experiencing the same issue but with varying differences in the description. But everyone is experiencing the same thing, we can connect to our codespace but and then are disconnected after a short period of time.

This is the project I am working out of https://github.com/talk2MeGooseman/stream_closed_captioner_phoenix. It is not using the default dev container but a configuration provided from the dev-container repo. 

Is there anything I can do to better debug this connectivity issue or provide better information from the log so the team can figure out why were experiencing connectivity problems?

**Comment by rodrigovaras** [comment]
Hi @talk2MeGooseman, we are investigating this issue.
Can you confirm that you are using the browser to connect to your codespace?
If yes, can you experiment with the vscode extension connecting to the same codespace?
The vscode has a command 'Export logs' that could help us identify the issue with more accuracy.

**Reply by talk2MeGooseman** [reply]
This screenshot is from the browser since I found it easier to display the logs. I am getting the issue on VSCode proper also and glad there is a way to export the logs. I'll do that shortly.

**Reply by talk2MeGooseman** [reply]
@rodrigovaras Good news! Whatever your all doing has solved the problem I have been having this is for the past couple of weeks. Amazing! So far I have been able to maintain a connection for a while now without any problems. Thanks for all the hard work.

**Comment by legomushroom** [comment, marked as answer]
@talk2MeGooseman are you still seeing the issues? We've shipped an update to the connection logic recently that should resolve such errors, mind installing the latest extension from the marketplace and see if it solves the problem for you? 🙏

**Reply by talk2MeGooseman** [reply]
I was using Codespaces for the past couple of days without any issues, nice, fast, and saved my battery life working remote. Everything is all good @legomushroom

**Reply by legomushroom** [reply]
Thanks for validating! 🙌

**Reply by DennesTorres** [reply]
I was using codespaces this evening and faced many issues that I'm blaming on instability:

- The frequent request for reconnect is annoying
- The terminal was hanging and gave me some trouble before starting
- The Live Server extension broke. The extension was working and out of the blue it stopped and gave me the message "Command 'Live Server: Open with Live Server' resulted in an error (command 'extension.liveServer.goOnline' not found)"
- The format document command asked me to install an HTML formatter, I installed but it was not recognized.

Kind Regards!

**Comment by Alexander-85vip** [comment]
> For the past couple of weeks, Codespaces has been impossible to work in. My already existing Codespace started to lose connection and attempt to reconnect constantly. To double-check that I didn't somehow mess up my container I went ahead and deleted my codespace and re-created it with the same issue. 
> 
> 
> 
> Here is a screenshot of one of the logs:
> 
> ![image](https://user-images.githubusercontent.com/1203718/115969061-e7eadd00-a4ef-11eb-8676-081053361af8.png)
> 
> 
> 
> I have also seen a couple of other issues in this discussion board with people experiencing the same issue but with varying differences in the description. But everyone is experiencing the same thing, we can connect to our codespace but and then are disconnected after a short period of time.
> 
> 
> 
> This is the project I am working out of https://github.com/talk2MeGooseman/stream_closed_captioner_phoenix. It is not using the default dev container but a configuration provided from the dev-container repo. 
> 
> 
> 
> Is there anything I can do to better debug this connectivity issue or provide better information from the log so the team can figure out why were experiencing connectivity problems?
> 
> 
- > ]()****

**Reply by legomushroom** [reply]
@Alexander-85vip @DennesTorres GH Codespaces logs would help to investigate this. 

1. the is the `Codespaces: Export Logs` command to get logs out of the codespace as a zip file
2. your client logs are in `{temp folder}/codespaces_logs` (`$TMPDIR/codespaces_logs` on Mac OSx)

Both are valuable but `2.` might work better in this case. Please note that the logs might have personal and sensitive information, so please don't attach the logs to the thread but DM them directly to me at `legomushroom@github.com`(or `olsolomk@microsoft.com`).

**Comment by vadehi-8** [comment]
my Codespaces is just never loading, always reconnecting and reloading the window please help me out
<img width="1592" height="656" alt="image" src="https://github.com/user-attachments/assets/65d3ab78-aefc-4427-a3c9-92d84cde44bd" />
<img width="1396" height="598" alt="image" src="https://github.com/user-attachments/assets/7baabea6-8057-4e86-bdca-73de4aa56a74" />

**Comment by sean-mcclure** [comment]
This is happening every time I try to work with this tool. It works for about 20 minutes and then freezes, and spends about another 20 minutes trying to reconnect. This has become a completely useless coding environment. It doesn't even save your uncommitted changes in cache (GitPod does this) so you lose everything not committed. This is really unacceptable. People have businesses they are trying to run.
